tests/components/recorder/test_models.py

```diff
@@ -165,3 +173,68 @@ def test_states_from_native_invalid_entity_id():
 
     state = state.to_native(validate_entity_id=False)
     assert state.entity_id == "test.invalid__id"
+
+
+async def test_process_timestamp():
+    """Test processing time stamp to UTC."""
+    datetime_with_tzinfo = datetime(2016, 7, 9, 11, 0, 0, tzinfo=dt.UTC)
+    datetime_without_tzinfo = datetime(2016, 7, 9, 11, 0, 0)
+    est = pytz.timezone("US/Eastern")
+    datetime_est_timezone = datetime(2016, 7, 9, 11, 0, 0, tzinfo=est)
+    nst = pytz.timezone("Canada/Newfoundland")
+    datetime_nst_timezone = datetime(2016, 7, 9, 11, 0, 0, tzinfo=nst)
+    hst = pytz.timezone("US/Hawaii")
+    datetime_hst_timezone = datetime(2016, 7, 9, 11, 0, 0, tzinfo=hst)
+
+    assert process_timestamp(datetime_with_tzinfo) == datetime(
+        2016, 7, 9, 11, 0, 0, tzinfo=dt.UTC
+    )
+    assert process_timestamp(datetime_without_tzinfo) == datetime(
+        2016, 7, 9, 11, 0, 0, tzinfo=dt.UTC
+    )
+    assert process_timestamp(datetime_est_timezone) == datetime(
+        2016, 7, 9, 15, 56, tzinfo=dt.UTC
+    )
+    assert process_timestamp(datetime_nst_timezone) == datetime(
+        2016, 7, 9, 14, 31, tzinfo=dt.UTC
+    )
+    assert process_timestamp(datetime_hst_timezone) == datetime(
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5737505)